File: dot-claude/hooks/statusline/gh.go
package main

import (
	"context"
	"os/exec"
	"strings"
	"time"
)

func getGhAccount() string {
	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
	defer cancel()
	cmd := exec.CommandContext(ctx, "gh", "auth", "status")
	out, _ := cmd.CombinedOutput()

	var current string
	for _, line := range strings.Split(string(out), "\n") {
		if idx := strings.Index(line, "Logged in to github.com account "); idx >= 0 {
			rest := line[idx+len("Logged in to github.com account "):]
			if fields := strings.Fields(rest); len(fields) > 0 {
				current = fields[0]
			}
		}
		if strings.Contains(line, "Active account: true") && current != "" {
			return current
		}
	}
	return ""
}

func ghSegment(cwd string) string {
	account := getGhAccount()
	if account == "" {
		return ""
	}

	seg := CPink + "\uf09b " + account + Rst

	gitUser := gitCmd(cwd, "config", "user.name")
	if gitUser != "" && !strings.EqualFold(gitUser, account) {
		seg += Sep + CCyan + "\ue702 " + gitUser + Rst
	}

	return seg
}

File: dot-claude/hooks/statusline/gh.js
// GitHub account segment — shows active gh auth account

const { execSync } = require('child_process');
const { RST } = require('./shared');

function ghSegment() {
  try {
    const out = execSync('gh auth status 2>&1', { encoding: 'utf8', timeout: 3000 });
    let current = null;
    for (const line of out.split('\n')) {
      const match = line.match(/Logged in to github\.com account (\S+)/);
      if (match) current = match[1];
      if (line.includes('Active account: true') && current) {
        return `\x1b[38;2;110;84;148m\uf09b ${current}${RST}`;
      }
    }
  } catch {}
  return '';
}

module.exports = { ghSegment };
